homeassistant/components/light/yeelight.py

```diff
@@ -286,13 +327,63 @@ def set_flash(self, flash) -> None:
             except BulbException as ex:
                 _LOGGER.error("Unable to set flash: %s", ex)
 
+    @_cmd
+    def set_effect(self, effect) -> None:
+        """Activate effect."""
+        if effect:
+            from yeelight import (Flow, BulbException)
+            from yeelight.transitions import (disco, temp, strobe, pulse,
+                                              strobe_color, alarm, police,
+                                              police2, christmas, rgb,
+                                              randomloop, slowdown)
+            # transition = int(self.config[CONF_TRANSITION])
+            if effect == EFFECT_DISCO:
+                flow = Flow(count=0, transitions=disco(),)
```

Comment:
Yes just passing the list with transitions does work, but that is what i originally had. 

@fabaff commented: 
> I think that the logic for the effects should be located in the yeelight module itself. This way the effects can be maintained at one place.

So this way we don't have to maintain the list for every transition, in case of changes in the `yeelight` module